### ISSUE TITLE
Catching BaseException is not a pylint issue where as catching Exception class is.

### ISSUE THREAD

**Issue by deepakmahapatra** [PyCQA/pylint#2741]
### Steps to reproduce
1. `try:
           your code placeholder
     except Exception
           do something
     excpet BaseException
           do something`
2. Running the pylint is not catching the 2nd block of Exception catching.

### Current behavior
catching BaseException is not a pylint issue.

### Expected behavior
It should be a warning as a default behavior.

### pylint --version output
pylint 2.2.2
astroid 2.1.0
Python 3.7.1 (default, Nov 28 2018, 11:51:54) 
[Clang 10.0.0 (clang-1000.11.45.5)]

**Comment by bgehman** [comment]
Catching too general of an Exception:
```
try:
    pass
except Exception:
    pass
```

Results in `W:  3, 7: Catching too general exception Exception (broad-except)`

Catching `BaseException`, which class `Exception` derives from (even more general):
```
try:
    pass
except BaseException:
    pass
```

No warning at all.

**Comment by PCManticore** [comment]
Thanks for the report @deepakmahapatra and @bgehman
